crates/sui-replay/src/replay.rs

```diff
@@ -1842,7 +1842,9 @@ impl ChildObjectResolver for LocalExec {
             receiving_object_id: &ObjectID,
             receive_object_at_version: SequenceNumber,
         ) -> SuiResult<Option<Object>> {
-            let recv_object = match self_.get_object(receiving_object_id)? {
+            let recv_object = match self_
+                .download_object_by_upper_bound(receiving_object_id, receive_object_at_version)?
```

Comment:
Just to double-check `download_object_by_upper_bound` will fetch an object with a version == upper_bound if one is available correct? If so LGTM.

Reply:
yeah, it's <= target_version

Reply:
Looking at it, it might be more efficient/better if instead of doing this (which should work) you could instead:
* At replay start when downloading objects, you can download the specific object version in the `transaction.receiving_objects()` 
* Here, you should then be able to call `get_object_by_key` and that should have the desired semantics (and in fact be a bit tighter semantically)

Reply:
oh great! let me check it and try that instead